src/modules/admin/components/formRules/FormRule.tsx

```diff
@@ -34,17 +30,29 @@ const nonProjectFormRoles = [FormRole.Organization];
 interface Props {
   rule: FormRuleFieldsFragment;
   formRole: FormRole;
+  formId: string;
 }
 
-const FormRule: React.FC<Props> = ({ rule, formRole }) => {
+const FormRule: React.FC<Props> = ({ rule, formRole, formId }) => {
+  // TODO - This duplicates some functionality from ProjectApplicabilitySummary used in the Project config table.
```

Comment:
To that point, it will be easier to refactor if the `useDeactivateFormRuleMutation` doesn't live here. (Since we don't need that for  Project config context).

Previously `FormRule` served as more of a presentational component, but we make it a data-fetching component by adding this mutation. I would suggest moving the mutation out into `FormRuleTable`, which is already doing data fetching. The `FormRule` component could accept an optional `actionButtons`(?) prop that we can use to pass the `<IconButton`. That makes it a little more flexible for use elsewhere, and keeps it "presentational".

Reply:
Makes sense to me. Thanks! This change causes all the deletion buttons to be greyed out during a mutation (instead of just the button pertaining to the rule that's getting deleted by this mutation.) I'm not sure if there was a way around this, I was checking the [useMutation docs](https://www.apollographql.com/docs/react/data/mutations/#usemutation-api) to see if there is an `onStarted` or similar callback that we could use to 'listen' to which ID is currently being deleted and then separately manage a piece of state for which one the `loading` applies to, but I don't think such a callback exists. It doesn't seem like a huge deal to me though, I'm fine with leaving it as-is.